src/ppom-controller.ts

```diff
@@ -828,11 +863,13 @@ export class PPOMController extends BaseControllerV2<
           Number(this.state.chainStatus[c2]?.lastVisited) -
           Number(this.state.chainStatus[c1]?.lastVisited),
       )[NETWORK_CACHE_LIMIT.MAX];
-      oldChaninIds.push(oldestChainId);
+      if (oldestChainId !== undefined) {
+        oldChainIds.push(oldestChainId);
+      }
```

Comment:
can we drop a `!` on oldestChainId above instead? We already check that the array is long enough. TS just doesn't believe us

Reply:
Sure thing, fixed here! [`da5a54f` (#89)](https://github.com/MetaMask/ppom-validator/pull/89/commits/da5a54f53fd61c6413b05fc34ac5fa0275255457) It's annoying when TypeScript refuses to be convinced.😅